### ISSUE TITLE
Confusion About Gaussian Position Calculation

### ISSUE THREAD

**Issue by johnren-code** [hustvl/GaussTR#18]
You let ref_pts multiply image_shape at 
```
points = torch.cat([
            ref_pts * torch.tensor(self.image_shape[::-1]).to(x),
            sample_depth * (1 + deltas[..., 2:3])
        ], -1)
```
And Then you let x, y of points multiply depth again there:
```
points = torch.cat(
            [points[..., :2] * points[..., 2:3], points[..., 2:3]], dim=-1)
```
Why do we need to multiply twice ?

**Comment by npurson** [comment]
```python
ref_pts * torch.tensor(self.image_shape[::-1]).to(x)
```  

scales the normalized reference points (`ref_pts` in [0,1] range) to pixel coordinates based on `image_shape`. 

```python
points[..., :2] * points[..., 2:3]
```  

converts the 2D image-plane coordinates into 3D space by incorporating depth (`points[..., 2:3]`). This follows the standard perspective projection model:

```
$ z_c \cdot (u, v, 1) = K \cdot P_c $
```

Please refer to camera transformations if you need more infomation.
